src/Services/AuthenticationService.php

```diff
@@ -154,26 +227,73 @@
 
         $state[State::RESTART] = $request->getUri()->__toString();
 
+        // Data required after we get back from a ProcessingChain redirect
+        $state['authorizationRequest'] = $authorizationRequest;
```

Comment:
Here League\OAuth2\Server\RequestTypes\AuthorizationRequest as OAuth2AuthorizationRequest is set, while in getAuthorizationRequestFromState() there is a check if the instance is of League\OAuth2\Server\RequestTypes\AuthorizationRequest. I think this may break Server/Grants/OAuth2ImplicitGrant.php, which only uses  OAuth2AuthorizationRequest (haven't tested it yet).

Reply:
i pushed a fixed and now the `AuthenticationService::getAuthorizationRequestFromState`  returns the correct object instance type.